### ISSUE TITLE
Error while fetching league list: Response code 403

### ISSUE THREAD

**Issue by kornuolis2** [PathOfBuildingCommunity/PathOfBuilding#8384]
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate open **and closed** issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues?q=is%3Aissue)

### What platform are you running Path of Building on?

Windows

### How is Path of Building expected to behave?

Clicking Trade for these items  i am supposed to enter my session ID,realm, league and go on looking for gear updates. 

### How does Path of Building behave?

In "Trade for these items" window i receive Error while fetching league list: Response code 403

Realm and league dropdown are inactive.
![image](https://github.com/user-attachments/assets/9c5c2b5a-ff7f-42a6-ba4e-8dc6a4b34724)

Portable version works only once, after unpacking. It still shows the error, but the message disappears after item search
![image](https://github.com/user-attachments/assets/db0905c9-e16c-4951-bcf7-cee79fb271d9)
After quiting portable version and opening it again the issue repeats itself as shown on the very first screenshot above.
Workaround is to unpack the zip into new folder or delete these 2 files:
![image](https://github.com/user-attachments/assets/00bbb57e-3ba4-4a31-828d-cf234d3cfad0)
but in this case i need to enter Username for importing and sessionID again.

### How to reproduce the issue

Import any charachters, switch to Items tab and click on Trade for these items.

**Comment by cemkod** [comment]
Having the same problem.

**Comment by LocalIdentity** [comment]
Your poesessid has expired and it's trying to use the old number to access the website and this causes and error. Should be fixed when I merge this PR https://github.com/PathOfBuildingCommunity/PathOfBuilding/pull/8386

**Comment by Since-T** [comment]
> Your poesessid has expired and it's trying to use the old number to access the website and this causes and error. Should be fixed when I merge this PR [#8386](https://github.com/PathOfBuildingCommunity/PathOfBuilding/pull/8386)
> 
> For now, update your poesessid or remove it

Got the same Problem, but for me, i tested both versions (portable and installed). Both dont work, even when entering a new processid i always get the error code 403.
I tested every workaround here and in the PR but nothing works for me.

**Comment by Kanaxai** [comment]
Same problem here, could be an issue with private leagues as I am trying to use the POB trade in Pohx league.

**Comment by w4fz5uck5** [comment]
> > Your poesessid has expired and it's trying to use the old number to access the website and this causes and error. Should be fixed when I merge this PR [#8386](https://github.com/PathOfBuildingCommunity/PathOfBuilding/pull/8386)
> > For now, update your poesessid or remove it
> 
> Got the same Problem, but for me, i tested both versions (portable and installed). Both dont work, even when entering a new processid i always get the error code 403. I tested every workaround here and in the PR but nothing works for me.
> 
> EDIT: Removing the processID will go into No Session Mode which will work fine to load the Leagues. But you cant load a private league anymore with that. Because every time you enter a processID the leagues can not get loaded.

To resolve this issue (including for the Pohx league), follow these steps:

1 - Remove your current POESESSID in PoB’s "Session mode" and save it as null.
2 - Re-enter your POESESSID and save the changes.
3 - If you're looking to find better items for your build, follow these steps:

Click "Find Best" and select your pre-defined weight sum filters.
1 - Open your browser and navigate to the Settlers league.
2 - Change the current league to Pohx.
3 - Browse for the best items to optimize your build.

**Comment by Since-T** [comment]
@w4fz5uck5 But with that you only use the "no session mode". Than you don't need to enter your POESESSID at all.
Than you can generate links without errors.

But the Feature to compare them as a list directly in pob and the feature that it try different weighed links doesn't work here. For that it need the private league trade website and the income from the offers that exist there.
With the "workaround" you describe you can only search with all items exist in a way bigger league/economy.

For private Leagues you cant use the "session mode" feature at all. All workaround just use the "no session mode" even when the text says other. This is what i want to point out.

**Comment by TheFinalChiTown** [comment]
I'm also having this same issue. I can temporarily work around it by deleting, saving null, then re-entering, and saving my POESESSID. However, the problem returns if I restart the application or load another build. League: Settlers.

**Comment by settonull** [comment]
I'm having the same issue still (2.52.3) - have to null the session id to get the league list.  Of course it's kinda mute because it won't generate meaningful searches at all. but I guess I should file a different bug report for that.

**Comment by delebota** [comment]
Same issue, tried reinstalling and it's still broken for me

**Comment by Loadst0ne** [comment]
Still happening in Phrecia

**Comment by Ham5terzilla** [comment]
Problem actual. Fix, please. Best POE tool. <3

**Comment by Protuhj** [comment]
Looking into this a little bit, from what I can tell, this is their CloudFare spam protection blocking the request.

If you look at your cookies when loading the https://www.pathofexile.com/trade site, you'll notice a `cf_clearance=` cookie in addition to the POESESSID.

All the code is trying to do is download the object at the bottom of the page, `require(["main"], function(){require(["trade"],...` and it seems like this is the only (?) page that **requires** the `cf_clearance` cookie?

I think if the download fails due to the 403, it should just fall back to the static list and/or allow us to paste in the `require(["main"], function(){require(["trade"],...` code so it can parse our personalized leagues, without needing to make that initial request.

**Comment by schrmm** [comment]
Yes, I can confirm it too. Very frustrating. Searching for timeless jewels also dont work for me.

**Comment by Nightblade** [comment]
Is this still happening with the latest version of PoB?

**Comment by settonull** [comment]
> Is this still happening with the latest version of PoB?

I just tested on 2.56.0 and it looked good to me - came up clean showing the leagues.  Once I updated the sessionid it seemed to be returning meaningful results (unlike before).

**Comment by Nightblade** [comment]
Closing as issue is probably resolved.

**Comment by TheFinalChiTown** [comment]
This error is happening to me again. Behavior is the same as in my comment on Feb 9, 2025:

"I'm also having this issue. I can temporarily work around it by deleting, saving null, then re-entering, and saving my POESESSID. However, the problem returns if I restart the application or load another build."

**Comment by vaisest** [comment]
> This error is happening to me again. Behavior is the same as in my comment on Feb 9, 2025:
> 
> "I'm also having this issue. I can temporarily work around it by deleting, saving null, then re-entering, and saving my POESESSID. However, the problem returns if I restart the application or load another build."

Yes this is a known newer issue. The tool accesses the trade site page directly which has some sort of Cloudflare thing that requires JS.

This is solved by either #9812 (nukes the mess that is poesessid altogether) or probably by #9908 or #9912
